File: frontend_admin/src/services/reportService.js
import api from './api';

export const reportService = {

    getRevenueSummary: async (fromDate, toDate) => {
        const response = await api.get('/admin/reports/revenue-summary', {
            params: { fromDate, toDate }
        });
        return response.data;
    },

    getRevenueByTime: async (fromDate, toDate, groupBy = 'MONTH') => {
        const response = await api.get('/admin/reports/revenue-by-time', {
            params: { fromDate, toDate, groupBy }
        });
        return response.data;
    },

    getRevenueByProduct: async (fromDate, toDate, limit = 20) => {
        const response = await api.get('/admin/reports/revenue-by-product', {
            params: { fromDate, toDate, limit }
        });
        return response.data;
    },
};

export default reportService;

File: backend/store-management/src/main/java/com/storemanagement/service/ReportService.java
package com.storemanagement.service;

import com.storemanagement.dto.report.RevenueByProductDTO;
import com.storemanagement.dto.report.RevenueByTimeDTO;
import com.storemanagement.dto.report.RevenueSummaryDTO;

import java.time.LocalDate;
import java.util.List;

public interface ReportService {

    RevenueSummaryDTO getRevenueSummary(LocalDate fromDate, LocalDate toDate);

    List<RevenueByTimeDTO> getRevenueByTime(LocalDate fromDate, LocalDate toDate, String groupBy);

    List<RevenueByProductDTO> getRevenueByProduct(LocalDate fromDate, LocalDate toDate, Integer limit);
}
